wasm-test-app/src/WASMTests.tsx

```diff
@@ -0,0 +1,212 @@
+
+        fetch('/data/test.provekey')
+            .then(res => res.blob())
+            .then(blob => setPkFile(new File([blob], 'test.provekey')));
+
+        fetch('/data/test.witness.json') // replace 'data' with your actual file name
+            .then(res => res.blob())
+            .then(blob => setDataFile(new File([blob], 'test.witness.json')));
+
+        fetch('/data/test.proof')
+            .then(res => res.blob())
+            .then(blob => setProofFile(new File([blob], 'test.proof')));
+
+        fetch('/data/test.key')
+            .then(res => res.blob())
+            .then(blob => setVkFile(new File([blob], 'test.key')));
+
+        fetch('/data/message')
+            .then(res => res.blob())
+            .then(blob => setMessageFile(new File([blob], 'message')));
+    }, []);
+
+    const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => { };
+    const handleGenPkButton = async (modelFile: File, srsFile: File, circuitSettingsFile: File) => {
+        try {
+            const model_ser = await readUploadedFileAsText(modelFile);
+            const srs_ser = await readUploadedFileAsText(srsFile);
+            const circuit_settings_ser = await readUploadedFileAsText(circuitSettingsFile);
+            const result_pk = gen_pk_wasm(model_ser, srs_ser, circuit_settings_ser);
+            console.log(result_pk)
+            setPkResult(`Test passed, ${result_pk}`);
+        } catch (error) {
+            setPkResult('GenPk test failed');
+            console.error(error);
+            throw new Error('GenPk test failed');
+        }
+    };
+    const handleGenVkButton = async (pkFile: File, circuitSettingsFile: File) => {
+        try {
+            const pk_ser = await readUploadedFileAsText(pkFile);
+            const circuit_settings_ser = await readUploadedFileAsText(circuitSettingsFile);
+            const result_vk = gen_vk_wasm(pk_ser, circuit_settings_ser);
+            console.log(result_vk)
+            setVkResult(`Test passed, ${result_vk}`);
+        } catch (error) {
+            setVkResult('GenVk test failed');
+            console.error(error);
+            throw new Error('GenVk test failed');
+        }
+    };
+
+    const handleGenProofButton = async (dataFile: File, pkFile: File, modelFile: File, circuitSettingsFile: File, srsFile: File) => {
+        try {
+            const data_prove_file = await readUploadedFileAsText(dataFile);
+            const pk_prove_file = await readUploadedFileAsText(pkFile);
+            const model_ser_prove_file = await readUploadedFileAsText(modelFile);
+            const circuit_settings_ser_prove_file = await readUploadedFileAsText(circuitSettingsFile);
+            const srs_ser_prove_file = await readUploadedFileAsText(srsFile);
+            const result_proof = prove_wasm(data_prove_file, pk_prove_file, model_ser_prove_file, circuit_settings_ser_prove_file, srs_ser_prove_file);
+            console.log(result_proof)
```

Comment:
IDK if this is your desired behavior. It's cleaner to show the user something in the UI than make them open their console to see stuff. If this is a feature let's be aware of this philosophy going forward. If this is a debug line, let's get it out of the codebase.